ahnlich/ai/src/engine/ai/mod.rs

```diff
@@ -0,0 +1,10 @@
+mod models;
+use ahnlich_types::keyval::{StoreInput, StoreKey};
+use models::ModelInfo;
+use std::num::NonZeroUsize;
+
+pub trait AIModelManager {
+    fn embedding_size(&self) -> NonZeroUsize;
```

Comment:
Can be a default implementation if it's already coming from the ModelInfo or get rid of it entirely even